lib/_http_outgoing.js

```diff
@@ -810,17 +810,18 @@ OutgoingMessage.prototype.end = function end(chunk, encoding, callback) {
     encoding = null;
   }
 
-  if (this.socket) {
-    this.socket.cork();
-  }
```

Comment:
I'm not sure this is ok. Does it mean that we will not cork() when ending without a response?

Reply:
Yes. The reason we cork is so that we write the response and eof in the same batch. If we don't have additional data corking here is not necessary.